### PR TITLE
504 update install scripts

### PR DESCRIPTION
**Pull request by tyronerees**
#### Description of Work

Relates #504 

#### Testing Instructions

1.
2.
3.

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?

### CODE REVIEW

File: build/build-scripts.sh [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+export CUTEST=${CUTEST_DIR}/cutest/ 
+export MYARCH="pc64.lnx.gfo" 
+export PYCUTEST_CACHE=${CUTEST_DIR}/pycutest_cache/
+export PYTHONPATH="${PYCUTEST_CACHE}:${PYTHONPATH}"
+cd $SIFDECODE
+printf "6\n2\n6\n" > sifdecode.input 
+printf "nny" >> sifdecode.input
+./install_sifdecode  < sifdecode.input
+cd $CUTEST
+printf "6\n2\n6\n2\n8\n" > cutest.input
```
[the remark this answers is not in the document]
Comment (AndrewLister-STFC) [reply build/build-scripts.sh:61]: It looks like this has been changing a lot recently... it's back to 8 again now...
@jfowkes suggests a custom bash script [here](https://jfowkes.github.io/pycutest/_build/html/install.html#installing-cutest-on-linux)

File: build/build-scripts.sh [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+rm mantid*.xz
+
+sudo apt-get install -y ccache
+ccache --max-size=20G
+
+mkdir -p Mantid
+
+cd Mantid
+cmake ${HERE}/mantid-6.3.0-Source -DENABLE_MANTIDPLOT=OFF
+make
```
Comment (AndrewLister-STFC) [inline build/build-scripts.sh:96]: This is what I did to get a working 6.2.0 -- I think it's just the `DENABLE_PRECOMMIT` that's important though. Also Sarah mentioned that Mantid is on 6.5.0 now??

We also found that numpy-1.21.6 works but numpy-1.23.4 does not for mantid-6.2.0
```
wget https://downloads.sourceforge.net/project/mantid/6.2/mantid-6.2.0-Source.tar.xz
tar -xvf mantid-6.2.0-Source.tar.xz
sudo mkdir /opt/Mantid
cd /opt/Mantid-6.2.0/
sudo cmake ~/mantid-6.2.0-Source/ -DENABLE_PRECOMMIT=OFF
sudo make
```

Comment (jfowkes) [reply build/build-scripts.sh:96]: @AndrewLister-STFC yes Mantid is on 6.5.0 but they now provide a prebuilt binary that should just work (no longer a need to compile from source), see https://www.mantidproject.org/installation/index.html#installation-full

Comment (AndrewLister-STFC) [reply build/build-scripts.sh:96]: 🙃 they used to do this and then stopped... hence why we had to do all this building

File: build/build-scripts.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (jfowkes) [reply build/build-scripts.sh:61]: @AndrewLister-STFC which is why we use an automated version for PyCUTEst:
https://jfowkes.github.io/pycutest/_build/html/install.html#installing-cutest-on-linux

File: build/build-scripts.sh [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+# Install pygsl
+sudo apt-get install -y gsl-bin libgsl-dev libgsl-dbg
+
+# Install RALFit
+sudo apt-get install -y gfortran lcov libblas-dev liblapack-dev libgdal-dev
+git clone https://github.com/ralna/RALFit
+pip install numpy
+mkdir -p ${HERE}/RALFit/libRALFit/build
+cd ${HERE}/RALFit/libRALFit/build
+cmake .. && make && make install
```
[the remark this answers is not in the document]
Comment (tyronerees) [reply build/build-scripts.sh:37]: cmake is installed via apt-get on line 10, so this should be ok

File: build/build-scripts.sh (hunk 1 above)
[the remark this answers is not in the document]
Comment (tyronerees) [reply build/build-scripts.sh]: I've changed this to Jari's script

File: build/build-scripts.sh (hunk 2 above)
Comment (jess-farmer) [reply build/build-scripts.sh:89]: I've also now got v6.6.0 running on python 3.10 using the above steps. I found that I had to remove the `qt5-default` and `clang-format-6.0` packages from the `sudo apt-get install` command and then also add `libeigen3-dev`. I also needed to pip install `versioningit`

**Comment by tyronerees** [comment]
The latest version of Mantid supports Python 3.10, so I am now using the binaries.  Adding `bin` and `lib` to the `PYTHONPATH` didn't work for me, but after some digging I found that Mantid's own python executable was adding these directories to the `PYTHONPATH`:
```
export PYTHONPATH=$PYTHONPATH:${MANTID_DIR}/bin/
export PYTHONPATH=$PYTHONPATH:${MANTID_DIR}/lib/python3.10/lib-dynload
export PYTHONPATH=$PYTHONPATH:${MANTID_DIR}/site-packages
export PYTHONPATH=$PYTHONPATH:${MANTID_DIR}/lib/python3.10/site-packages
export PYTHONPATH=$PYTHONPATH:${MANTID_DIR}/scripts
export PYTHONPATH=$PYTHONPATH:${MANTID_DIR}/lib/python3.10.zip
export PYTHONPATH=$PYTHONPATH:${MANTID_DIR}/lib/python3.10
```
I'm not sure all these are needed, but fitbenchmarking works with the binary release of Mantid on linux with all these set 🚀

**Comment by tyronerees** [comment]
I think we should be using Ansible for this, not building a bash script: see https://stfc-cloud-docs.readthedocs.io/en/latest/howto/TerraformAnsible.html

The work here should be merged/replaced by Andrew's iDAaaS work with Ansible

**Comment by tyronerees** [comment]
We should close this PR and create a new one that removes install scripts. Instead, we should document pointers to the iDAaaS ansible scripts, being clear what the expected available software is (e.g., Mantid and Matlab).

**Comment by AndrewLister-STFC** [comment]
Decided to go with ansible. See linked issue

**Pull request closed without merging**